CMakeLists.txt

```diff
@@ -108,7 +108,7 @@ endif()
 set(CMAKE_REQUIRED_FLAGS "-Wl,--as-needed")
 CHECK_CXX_SOURCE_COMPILES("int main() {return 0;}" LINKER_SUPPORTS_WLASNEEDED)
 if(LINKER_SUPPORTS_WLASNEEDED)
-        list(APPEND EXTRA_LINKER_FLAGS "-Wl,--as-needed")
+        SET(EXTRA_LINKER_FLAGS "-Wl,--as-needed")
```

Comment:
should append, not replace.

Reply:
This is our own variable which is not set above ?
I tested the following :
`SET(EXTRA_LINKER_FLAGS "${EXTRA_LINKER_FLAGS} -Wl,--as-needed")`
But then CMake complains because EXTRA_LINKER_FLAGS begins with a space !

Reply:
ok, that makes sense.  Thanks.